plugins/workbench-impl/src/components/navigator/SpacesNav.svelte

```diff
@@ -46,12 +47,19 @@
       showModal(model.createComponent, {})
     }
   }
+
+  function selectSpace(id: Ref<Space>) {
+    const loc = getCurrentLocation()
+    loc.path[2] = id
```

Comment:
I'm personally not like this approach, it is not flexible, how it work if not enought arguments are provided?

Reply:
I do not like this too, please suggest anything to make it good. Also I do not understand what you mean by "not enought arguments are provided"?

Reply:
How we could be sure we have at least 2 values in loc.path?

Reply:
Routes stuff you removed was my proposal :-)

Reply:
Let's discuss this on meeting, I tried to use routes stuff during weekend but it does not work well for me